evmrpc/block.go

```diff
@@ -31,27 +32,33 @@ func NewBlockAPI(tmClient rpcclient.Client, k *keeper.Keeper, ctxProvider func(i
 	return &BlockAPI{tmClient: tmClient, keeper: k, ctxProvider: ctxProvider, txConfig: txConfig}
 }
 
-func (a *BlockAPI) GetBlockTransactionCountByNumber(ctx context.Context, number rpc.BlockNumber) *hexutil.Uint {
+func (a *BlockAPI) GetBlockTransactionCountByNumber(ctx context.Context, number rpc.BlockNumber) (result *hexutil.Uint, returnErr error) {
+	startTime := time.Now()
+	defer recordMetrics("eth_GetBlockTransactionCountByNumber", startTime, returnErr == nil)
 	numberPtr, err := getBlockNumber(ctx, a.tmClient, number)
 	if err != nil {
-		return nil
+		return nil, err
 	}
 	block, err := blockWithRetry(ctx, a.tmClient, numberPtr)
 	if err != nil {
-		return nil
+		return nil, err
 	}
-	return a.getEvmTxCount(block.Block.Txs)
+	return a.getEvmTxCount(block.Block.Txs), nil
 }
 
-func (a *BlockAPI) GetBlockTransactionCountByHash(ctx context.Context, blockHash common.Hash) *hexutil.Uint {
+func (a *BlockAPI) GetBlockTransactionCountByHash(ctx context.Context, blockHash common.Hash) (result *hexutil.Uint, returnErr error) {
+	startTime := time.Now()
+	defer recordMetrics("eth_GetBlockTransactionCountByHash", startTime, returnErr == nil)
 	block, err := blockByHashWithRetry(ctx, a.tmClient, blockHash[:])
 	if err != nil {
-		return nil
+		return nil, err
 	}
-	return a.getEvmTxCount(block.Block.Txs)
+	return a.getEvmTxCount(block.Block.Txs), nil
 }
 
-func (a *BlockAPI) GetBlockByHash(ctx context.Context, blockHash common.Hash, fullTx bool) (map[string]interface{}, error) {
+func (a *BlockAPI) GetBlockByHash(ctx context.Context, blockHash common.Hash, fullTx bool) (result map[string]interface{}, returnErr error) {
+	startTime := time.Now()
+	defer recordMetrics("eth_GetBlockByHash", startTime, returnErr == nil)
```

Comment:
small nit, if we could keep the case of the eth spec which is like `eth_getBlockByHash` it may be easier for us

Reply:
make sense